### PR TITLE
Don't retry invalid request to openai

### PR DESCRIPTION
**Pull request by collindutter**
Draft PR for #148

### CODE REVIEW

**Review by vasinov — COMMENTED** [review]
We could add `ignored_exception_types: List[Type[Exception]]` to `ExponentialBackoffMixin` and set that in each subclass instead.

**Comment by collindutter** [comment]
I like that more, will update tomorrow.

File: griptape/drivers/prompt/openai_completion_prompt_driver.py [hunk 1]
```diff
@@ -33,6 +34,7 @@ class OpenAiCompletionPromptDriver(BasePromptDriver):
         kw_only=True
     )
     user: str = field(default="", kw_only=True)
+    ignored_exception_types: Tuple[Type[Exception], ...] = field(default=(openai.InvalidRequestError), kw_only=True)
```
Comment (vasinov) [inline griptape/drivers/prompt/openai_completion_prompt_driver.py:37]: This default needs to be wrapped in a `Factory`.

File: griptape/drivers/prompt/openai_chat_prompt_driver.py [hunk 2]
```diff
@@ -33,6 +34,7 @@ class OpenAiChatPromptDriver(BasePromptDriver):
         kw_only=True
     )
     user: str = field(default="", kw_only=True)
+    ignored_exception_types: Tuple[Type[Exception], ...] = field(default=(openai.InvalidRequestError), kw_only=True)
```
Comment (vasinov) [inline griptape/drivers/prompt/openai_chat_prompt_driver.py:37]: This default needs to be wrapped in a `Factory`.

File: griptape/mixins/exponential_backoff_mixin.py [hunk 3]
```diff
@@ -13,13 +14,15 @@ class ExponentialBackoffMixin(ABC):
         default=lambda s: logging.warning(s),
         kw_only=True
     )
+    ignored_exception_types: Tuple[Type[Exception], ...] = field(default=(), kw_only=True)
```
Comment (vasinov) [inline griptape/mixins/exponential_backoff_mixin.py:17]: Should be `factory=tuple`.

**Review by vasinov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**